### PR TITLE
fix: update notation-go

### PR DESCRIPTION
**Pull request by JeyJeyGao**
This PR updates notation-go to v0.0.0-20220809210532-f0a54093ba32.
1. update directory structure logic: if config file doesn't exist, create the file at user config directory instead of the system directory;
2. dir.Path.Localkey signature has been changed and we need to adapt our code.

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Comment by dtzar** [comment]
Depends on merge of https://github.com/notaryproject/notation-go/pull/104

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/notation/cert_gen.go [hunk 1]
```diff
@@ -43,14 +43,14 @@ func generateTestCert(opts *certGenerateTestOpts) error {
 	fmt.Println("generated certificates expiring on", rsaLeafCertTuple.Cert.NotAfter.Format(time.RFC3339))
 
 	// write private key
-	keyPath := dir.Path.Localkey(name, dir.KeyExtension)
+	keyPath := dir.Path.Localkey(name, dir.LocalKeyExtension)
```
Comment (rgnote) [inline cmd/notation/cert_gen.go:46]: Update this according to https://github.com/notaryproject/notation-go/pull/104?

Comment (JeyJeyGao) [reply cmd/notation/cert_gen.go]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by qmuntal** [comment]
I would argue against merging any PR which does not have an appropriate description, even if the changes are small and self-documenting.

For example, this PR could say something like:

```
This PR updates `notation-go` to v0.0.0-20220809210532-f0a54093ba32.

The new version does not contain any known bug fix, but `dir.Path.Localkey` signature has been changed and we need to adapt our code.
```

**Comment by dtzar** [comment]
@rgnote - your feedback has been addressed.  Can you please take a quick look?  This is the only PR we need to finish alpha.3 release.
@qmuntal - I went ahead and added your description to the PR.

**Review by qmuntal — APPROVED** [review]

**Review by rgnote — COMMENTED** [review]
LGTM

**Pull request merged**